packages/puppeteer-core/src/common/bidi/Realm.ts

```diff
@@ -46,6 +46,17 @@ export class Realm extends EventEmitter {
 
   setFrame(frame: Frame): void {
     this.#frame = frame;
+
+    // TODO(jrandolf): We should try to find a less brute-force way of doing
+    // this.
+    this.connection.on(
+      Bidi.ChromiumBidi.Script.EventNames.RealmDestroyed,
+      async () => {
+        const util = this.internalPuppeteerUtil;
+        delete this.internalPuppeteerUtil;
+        void (await util)?.dispose().catch(() => {});
```

Comment:
We should send it to debugError.

Reply:
We don't want to do this because we assume that it can error here.

Reply:
`.dispose().catch(debugError);` why would it be a problem to do this?

Reply:
It'd be noisy because we know that the utility can error here.